### ISSUE TITLE
Client generator generates code files with invalid/broken imports

### ISSUE THREAD

**Issue by sbstp** [go-swagger/go-swagger#2609]
## Problem statement

The client generator has begun generated invalid code, pretty much out of the blue, and I can't figure out what's going on. It generates a client file that imports stuff that it doesn't need and also does not import the operations package, creating an error.

The reason why I say this is out of the blue is because I've only just begun experiencing this error. I managed to generate a working client before with v0.27.0, with the same specs. For about a week now the client generator no longer works.

![image](https://user-images.githubusercontent.com/862368/126397443-9e9013bc-d860-45f1-837b-34b09a09d832.png)

![image](https://user-images.githubusercontent.com/862368/126397480-ed097831-ed07-4a99-8ae1-dd8ca620dfa2.png)

## Swagger specification
```json
{
  "host": "secure.na1.adobesign.com",
  "basePath": "/api/rest/v6",
  "schemes": [
    "https"
  ],
  "swagger": "2.0",
  "info": {
    "title": "transientDocuments",
    "version": "6.0.0"
  },
  "paths": {
    "/transientDocuments": {
      "post": {
        "produces": [
          "application/json"
        ],
        "operationId": "createTransientDocument",
        "summary": "Uploads a document and obtains the document's ID.",
        "description": "The document uploaded through this call is referred to as transient since it is available only for 7 days after the upload. The returned transient document ID can be used in the API calls where the uploaded file needs to be referred. The transient document request is a multipart request consisting of three parts - filename, mime type and the file stream. You can only upload one file at a time in this request.",
        "consumes": [
          "multipart/form-data"
        ],
        "parameters": [
          {
            "in": "header",
            "description": "An <a href=\"#\" onclick=\"this.href=oauthDoc()\" oncontextmenu=\"this.href=oauthDoc()\" target=\"oauthDoc\">OAuth Access Token</a> with any of the following scopes:<ul><li style='list-style-type: square'><a href=\"#\" onclick=\"this.href=oauthDoc('agreement_write')\" oncontextmenu=\"this.href=oauthDoc('agreement_write')\" target=\"oauthDoc\">agreement_write</a></li><li style='list-style-type: square'><a href=\"#\" onclick=\"this.href=oauthDoc('agreement_sign')\" oncontextmenu=\"this.href=oauthDoc('agreement_sign')\" target=\"oauthDoc\">agreement_sign</a></li><li style='list-style-type: square'><a href=\"#\" onclick=\"this.href=oauthDoc('widget_write')\" oncontextmenu=\"this.href=oauthDoc('widget_write')\" target=\"oauthDoc\">widget_write</a></li><li style='list-style-type: square'><a href=\"#\" onclick=\"this.href=oauthDoc('library_write')\" oncontextmenu=\"this.href=oauthDoc('library_write')\" target=\"oauthDoc\">library_write</a></li></ul>in the format <b>'Bearer {accessToken}'.",
            "name": "Authorization",
            "required": true,
            "type": "string"
          },
          {
            "in": "header",
            "description": "The userId or email of API caller using the account or group token in the format <b>userid:{userId} OR email:{email}.</b> If it is not specified, then the caller is inferred from the token.",
            "name": "x-api-user",
            "required": false,
            "type": "string"
          },
          {
            "in": "header",
            "description": "The userId or email in the format <b>userid:{userId} OR email:{email}.</b> of the user that has shared his/her account",
            "name": "x-on-behalf-of-user",
            "required": false,
            "type": "string"
          },
          {
            "in": "formData",
            "description": "A name for the document being uploaded. Maximum number of characters in the name is restricted to 255.",
            "name": "File-Name",
            "required": false,
            "type": "string"
          },
          {
            "in": "formData",
            "description": "The mime type of the document being uploaded. If not specified here then mime type is picked up from the file object. If mime type is not present there either then mime type is inferred from file name extension.",
            "name": "Mime-Type",
            "required": false,
            "type": "string"
          },
          {
            "in": "formData",
            "description": "The file part of the multipart request for document upload. You can upload only one file at a time.",
            "name": "File",
            "required": true,
            "type": "file"
          }
        ],
        "responses": {
          "200": {
            "description": "No response was specified",
            "schema": {
              "$ref": "#/definitions/TransientDocumentResponse"
            }
          }
        },
        "security": [
          {
            "oauth2": [
              "agreement_write",
              "agreement_sign",
              "widget_write",
              "library_write"
            ]
          }
        ]
      }
    }
  },
  "definitions": {
    "TransientDocumentResponse": {
      "required": [
        "transientDocumentId"
      ],
      "properties": {
        "transientDocumentId": {
          "type": "string",
          "description": "The unique identifier of the uploaded document that can be used in an agreement or a bulk send or web form creation call"
        }
      }
    }
  }
}
```

## Steps to reproduce
I run use the docker image to generate the client, the command looks like `docker run --rm -it --user 1000:1000 -v /home/simon:/home/simon -w /home/simon/projects/nestoca/adobe-sign/swagger quay.io/goswagger/swagger:v0.27.0 generate client -f /home/simon/projects/nestoca/adobe-sign/swagger/2.0/transientDocuments.json -t /home/simon/projects/nestoca/adobe-sign/swagger/pkg/transientDocuments`

## Environment
swagger version: v0.27.0 , v0.26.1
go version: 1.16.5
OS:  Debian 10 amd64

**Comment by sbstp** [comment]
Here's the log when I run generate client with v0.27.0:

```
2021/07/20 21:21:34 validating spec /home/simon/projects/nestoca/adobe-sign/swagger/2.0/transientDocuments.json
2021/07/20 21:21:35 preprocessing spec with option:  minimal flattening
2021/07/20 21:21:35 building a plan for generation
2021/07/20 21:21:35 generation target /home/simon/projects/nestoca/adobe-sign/swagger/pkg/transientDocuments
2021/07/20 21:21:35 planning definitions (found: 1)
2021/07/20 21:21:35 planning operations (found: 1)
2021/07/20 21:21:35 grouping operations into packages (packages: 1)
2021/07/20 21:21:35 operations for package packages "operations" (found: 1)
2021/07/20 21:21:35 planning meta data and facades
2021/07/20 21:21:35 rendering 1 templates for model TransientDocumentResponse
2021/07/20 21:21:35 name field TransientDocumentResponse
2021/07/20 21:21:35 package field models
2021/07/20 21:21:35 creating generated file "transient_document_response.go" in "/home/simon/projects/nestoca/adobe-sign/swagger/pkg/transientDocuments/models" as definition
2021/07/20 21:21:35 executed template asset:model
2021/07/20 21:21:35 rendering 2 templates for operation 
2021/07/20 21:21:35 name field createTransientDocument
2021/07/20 21:21:35 package field operations
2021/07/20 21:21:35 creating generated file "create_transient_document_parameters.go" in "/home/simon/projects/nestoca/adobe-sign/swagger/pkg/transientDocuments/client/operations" as parameters
2021/07/20 21:21:35 executed template asset:clientParameter
2021/07/20 21:21:35 name field createTransientDocument
2021/07/20 21:21:35 package field operations
2021/07/20 21:21:35 creating generated file "create_transient_document_responses.go" in "/home/simon/projects/nestoca/adobe-sign/swagger/pkg/transientDocuments/client/operations" as responses
2021/07/20 21:21:35 executed template asset:clientResponse
2021/07/20 21:21:35 rendering 1 templates for operation group 
2021/07/20 21:21:35 name field operations
2021/07/20 21:21:35 creating generated file "operations_client.go" in "/home/simon/projects/nestoca/adobe-sign/swagger/pkg/transientDocuments/client/operations" as client
2021/07/20 21:21:35 executed template asset:clientClient
2021/07/20 21:21:35 rendering 1 templates for application TransientDocuments
2021/07/20 21:21:35 name field TransientDocuments
2021/07/20 21:21:35 package field client
2021/07/20 21:21:35 creating generated file "transient_documents_client.go" in "/home/simon/projects/nestoca/adobe-sign/swagger/pkg/transientDocuments/client" as facade
2021/07/20 21:21:35 executed template asset:clientFacade
2021/07/20 21:21:36 Generation completed!

For this generation to compile you need to have some packages in your GOPATH:

	* github.com/go-openapi/errors
	* github.com/go-openapi/runtime
	* github.com/go-openapi/runtime/client
	* github.com/go-openapi/strfmt

You can get these now with: go get -u -f pkg/transientDocuments/...
```

**Comment by youyuanwu** [comment]
What is your GOPATH and do you have a go module?

**Comment by sbstp** [comment]
GOPATH is not provided to the container. The go module is `github.com/nestoca/adobe-sign`

**Comment by MJjainam** [comment]
I have seen the same issue. I used goimports binary to add/remove required go packages. 
`goimports -w *.go`

Goimports : https://pkg.go.dev/golang.org/x/tools/cmd/goimports

**Comment by youyuanwu** [comment]
What is the output of running "go get ./..." and "go vet ./..." in go mod folder?

**Comment by casualjim** [comment]
duplicate of #2608 

go imports got a fix recently golang/go#45398. I'll package a new release this weekend.

**Comment by sbstp** [comment]
Seems fixed in 0.28.0

**Comment by immunochomik** [comment]
I have the same problem and it is still there in 0.28.0 from some experiments it looks like content of imports in this generated file is dependent on what packages I have in already GOPATH. and this is very wired behaviour.  Basically if I have the client package in the GOPATH for example if imported by some other project, then the code generation works correctly but if it is not in GOPATH then the client/operations package will not be imported
